spec/controllers/trn_submissions_controller_spec.rb

```diff
@@ -7,31 +7,71 @@
 
   describe "#create" do
     let(:current_user) { create(:user) }
-    let(:trainee) { create(:trainee, :completed, provider: current_user.provider) }
 
     before do
       allow(controller).to receive(:current_user).and_return(current_user)
     end
 
     context "when all sections are completed" do
-      it "queues a background job to register trainee for TRN" do
-        expect {
+      context "and the course start date is in the future" do
+        let(:trainee) do
+          create(
+            :trainee,
+            :completed,
+            :with_study_mode_and_future_course_dates,
+            provider: current_user.provider,
+          )
+        end
+
+        it "queues a background job to register trainee for TRN" do
+          expect {
+            post :create, params: { trainee_id: trainee }
+          }.to have_enqueued_job(Dttp::RegisterForTrnJob).with(trainee, current_user.dttp_id)
+        end
+
+        it "queues a background job to poll for the trainee's TRN" do
+          expect(Dttp::RetrieveTrnJob).to receive(:perform_with_default_delay).with(trainee)
```

Comment:
nitpick: I think it might be nicer to move these 3 tests to a unit spec file for the `Trainees::SubmitForTrn` service, and over here expect that we call `Trainees::SubmitForTrn` depending.

Reply:
i thought the same thing, but ran out of steam - i will do it now!